frontend/src/static/js/v_summary_charts.js

```diff
@@ -260,6 +260,35 @@ window.vSummaryCharts = {
     getGroupTotalContribution(group) {
       return group.reduce((accContribution, user) => accContribution + user.totalCommits, 0);
     },
+
+    getGroupName(group) {
+      switch (this.filterGroupSelection) {
+      case 'groupByRepos':
+        return group[0].repoName;
+      case 'groupByAuthors':
+        return group[0].name;
+      default:
+        return '';
+      }
+    },
+
+    isGroupMerged(groupName) {
+      if (this.mergedGroups === 'all') {
+        return true;
+      }
+      if (this.mergedGroups === 'none') {
+        return false;
+      }
+      return this.mergedGroups.split(',').includes(groupName);
+    },
+
+    handleMergeGroup(groupName) {
```

Comment:
no longer need this method and the parent's `handleMergeGroup` as modifying the `group name -> isMerge` object here will affect the parent's as well.

Reply:
Ignore this, this may not be able to achievable as vue doesn't allow child component to update the parent's data. Using `$emit` is the only way. Perhaps when we transition to `vuex` then we won't be needing this anymore.